### ISSUE TITLE
irregular mask train the model?

### ISSUE THREAD

**Issue by Jason-xin** [JiahuiYu/generative_inpainting#89]
Hello, sorry to bother you...
You said it can not solve irregular mask in the version(#77)，you v2 paper is difficult for me... 
So, if I add irregular mask to your v1 model when training, it may be works well or not ?

What's more, when training Places2 model, all High-resolution images were used? It has 105G ...
If I want to train my own dataset with the given Places2 pre-trained model given, how many images and size(256*256 mentioned in paper) are required? 
![image](https://user-images.githubusercontent.com/13817995/42803032-430f86e0-89d7-11e8-915f-0d51443f304d.png)

**Comment by JiahuiYu** [comment]
I use the high-resolution images, which is important. During training, the code will random crop a 256x256 patch in high-resolutions images.

Adding irregular mask to DeepFill v1 training will definitely improves the results. But it is not a fundamental solution for free-form masks. Thus I would still suggest to understand DeepFill V2, especially the gated convolution. If you still have questions after carefully reading the paper, just ask them here and I will try my best to help.

If you want to train your own dataset, more images will be better. For all experiments, at least I use 30,000 images (CelebA-HQ model). You could have a reference at issue #88. They have fairly good results in most cases.
